hugr-core/src/types.rs

```diff
@@ -4,7 +4,7 @@ mod check;
 pub mod custom;
 mod poly_func;
 mod row_var;
-mod serialize;
+pub(crate) mod serialize;
```

Comment:
I leave it up to you, but you could keep the stronger encapsulation that comes with a private `serialize` module by moving `impl TryFrom<SerialSum> for Sum` from `hugr-core/src/ops/constant.rs` into `hugr-core/src/types/serialize.rs`